### DISCUSSION TITLE
[Question][Hw5] Define lattices

### DISCUSSION THREAD

**Discussion by Jaegoomon** [prosyslab-classroom/cs348-information-security#590] [Q&A]
Given type is `Bot`, `Neg`, `Zero`, `Pos`, and `Top`, do we have to define addtional layer of lattices on myself?

I mean according to the class, Upper the layer of `Neg`, `Zero`, `Pos` we can define another GLB such as ZP, ZN ...

Do I need to define those kinds of new types that include multitple given types(i.e ZP = Zero U Pos) in thise project?

**Comment by RiceBiscuits** [comment]
You can use already defined domain without adding additional types.
It is not necessary.
But please notice grading will be conducted based on current abstract domain.
please see *.expected files. Result of your analyzer should be matched those file's contents.
Otherwise, you cannot get score.
